### PR TITLE
DeFiZap user form

### PR DESCRIPTION
**Pull request by blurpesec**
### Description:
User form for defi zap. first step in the flow.

### ToDo:
- [x] Get assets from Figma for Bullish vs Bearish indicator.
- [x] Add them to the banner on user form. Add them to user education panel too.

### Screenshots:
image

### CODE REVIEW

File: common/v2/features/DeFiZap/components/DashboardZapCTA.tsx [hunk 1]
```diff
@@ -61,17 +46,10 @@ const DashboardZapCTA = ({ className }: any) => {
...
-      <DefiZapLogoImage src={defizaplogo} />
-      <DefiZapLogoText>{headingRight}</DefiZapLogoText>
-    </DefiZapLogo>
-  );
-
   return (
     <DashboardPanel
       heading={'Put your ETH to work for you'}
-      headingRight={RightHeading}
+      headingRight={DeFiZapLogo()}
```
Comment (FrederikBolding) [inline common/v2/features/DeFiZap/components/DashboardZapCTA.tsx]: Can we use it as a tag instead? Just seems prettier 😄

File: common/v2/features/DeFiZap/components/ZapForm.tsx [hunk 2]
```diff
@@ -1,34 +1,97 @@
...
+  &:disabled {
+    background-color: rgba(0, 122, 153, 0.3);
+  }
+`;
+
+const DeFiZapLogoContainer = styled.div`
+  margin-top: ${SPACING.BASE};
 `;
 
 export const EtherUUID = '356a192b-7913-504c-9457-4d18c28d46e6';
```
Comment (FrederikBolding) [inline common/v2/features/DeFiZap/components/ZapForm.tsx]: Isn't there a better way to do this?

File: common/v2/features/DeFiZap/components/ZapForm.tsx [hunk 3]
```diff
@@ -40,85 +103,116 @@ const ZapForm = (props: Props) => {
...
+              </FormFieldItem>
+              <FormFieldSubmitButton
+                type="submit"
+                onClick={() => {
+                  if (isValid) {
+                    onComplete(values);
+                  }
+                }}
+              >
+                Continue on!
```
Comment (FrederikBolding) [inline common/v2/features/DeFiZap/components/ZapForm.tsx]: Use translation

File: common/v2/features/DeFiZap/components/ZapIndicators.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import { SPACING, COLORS } from 'v2/theme';
+
+import bullishIndicator from 'assets/images/defizap/indicators/bullishIndicator.svg';
+import bearishIndicator from 'assets/images/defizap/indicators/bearishIndicator.svg';
+
+const IContainer = styled.div`
+  display: flex;
+`;
+
+const IImage = styled.img`
```
Comment (FrederikBolding) [inline common/v2/features/DeFiZap/components/ZapIndicators.tsx]: We usually go with `SImage` or simply `Image` if that makes more sense.

Nitpicky, sorry, but good to have a general consensus on this.

File: yarn.lock [hunk 5]
```diff
@@ -17154,7 +17154,7 @@ storybook-chromatic@^2.2.2:
     tree-kill "^1.1.0"
     uuid "^3.3.2"
 
-storybook-react-router@^1.0.8:
+storybook-react-router@1.0.8:
```
Comment (FrederikBolding) [inline yarn.lock]: ??

File: common/v2/features/DeFiZap/components/ZapSelectedBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        <BannerSubItemText>
+          <div>{zapSelected.title}</div>
+          <ZapNameGrey>{zapSelected.name}</ZapNameGrey>
+          <IndicatorItem />
+        </BannerSubItemText>
+      </BannerSubItem>
+      <BannerSubItem>
+        <BannerSubItemLabel>{translateRaw('PLATFORMS')}</BannerSubItemLabel>
+        <BannerSubItemText>
+          <ProtocolTagsList platformsUsed={zapSelected.platformsUsed} />
```
Comment (FrederikBolding) [inline common/v2/features/DeFiZap/components/ZapSelectedBanner.tsx]: Is there a reason these don't start with a capital letter?

File: common/v2/features/DeFiZap/components/ZapForm.tsx (hunk 2 above)
Comment (FrederikBolding) [reply common/v2/features/DeFiZap/components/ZapForm.tsx]: To clarify, I'm talking about the UUID line.

File: common/v2/features/DeFiZap/components/DashboardZapCTA.tsx (hunk 1 above)
Comment (blurpesec) [reply common/v2/features/DeFiZap/components/DashboardZapCTA.tsx]: > Can we use it as a tag instead? Just seems prettier smile

What do you mean by "tag"?

File: common/v2/features/DeFiZap/components/ZapForm.tsx (hunk 2 above)
Comment (blurpesec) [reply common/v2/features/DeFiZap/components/ZapForm.tsx]: I couldn't come up with anything better, you have any ideas?

File: yarn.lock (hunk 5 above)
Comment (blurpesec) [reply yarn.lock]: Just did a `sudo rm -rf node_modules/ && yarn` to update deps.

File: common/v2/features/DeFiZap/components/ZapForm.tsx (hunk 3 above)
Comment (blurpesec) [reply common/v2/features/DeFiZap/components/ZapForm.tsx]: We're going to go back over the whole flow and do translations at the end of it when we can have design review copy all in one go c:

File: common/v2/features/DeFiZap/components/ZapSelectedBanner.tsx (hunk 6 above)
Comment (blurpesec) [reply common/v2/features/DeFiZap/components/ZapSelectedBanner.tsx]: I think the design actually has them as all chars uppercase, so ill go ahead and do that.

**Comment by blurpesec** [comment]
> Small changes and questions.
> 
> Couldn't find the design so haven't checked if it matches completely.

https://www.figma.com/file/BY0SWc75teEUZzws8JdgLMpy/%5BMyCrypto%5D-GAU-Master?node-id=6189%3A113839

File: common/v2/features/DeFiZap/components/ZapIndicators.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+const SImage = styled.img`
+  height: 16px;
+  width: 16px;
+`;
+const SText = styled.p`
+  padding-left: ${SPACING.XS};
+  color: ${COLORS.PURPLE};
+  margin-bottom: 0px;
+  font-size: 16px;
```
Comment (FrederikBolding) [inline common/v2/features/DeFiZap/components/ZapIndicators.tsx]: Can we use `Typography` for text and `FONT_SIZE` constant for font-sizes?
There may be more cases than this one

File: common/v2/features/DeFiZap/components/ZapSelectedBanner.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  margin-right: ${SPACING.XS};
+  line-height: 16px;
+`;
+
+const BannerSubItemText = styled.div`
+  display: flex;
+  flex: 1;
+  justify-content: space-between;
+  margin-bottom: 0px;
+  line-height: 16px;
```
Comment (FrederikBolding) [inline common/v2/features/DeFiZap/components/ZapSelectedBanner.tsx]: Possibly use a constant here as well (among other places in this file)

File: common/v2/features/DeFiZap/components/ProtocolTagsList.tsx [hunk 9]
```diff
@@ -8,7 +8,7 @@ interface Props {
 
 const ProtocolTagsList = ({ platformsUsed }: Props) => {
   const platformTags = platformsUsed.map(platform => ({
-    tagText: platform,
+    tagText: platform.toUpperCase(),
     color: getPlatformColor(platform)
```
Comment (FrederikBolding) [inline common/v2/features/DeFiZap/components/ProtocolTagsList.tsx]: Compound color is off - it should be a light green (`#B3DD87`) according to the design.

File: common/v2/features/DeFiZap/components/ZapIndicators.tsx (hunk 7 above)
Comment (FrederikBolding) [reply common/v2/features/DeFiZap/components/ZapIndicators.tsx]: This should be bold as well it seems

**Pull request merged**